### PR TITLE
Ana/fix rewards updating

### LINKED ISSUE #3095: withdrawing rewards doesn't update available balance until refresh
😞

### PR DESCRIPTION
**Pull request by Bitcoinera**
Closes #3095 

**Description:**

I am getting close to fix this issue: making the rewards update by themselves, without refreshing. 
The most significant changes for this are found in the API side. I will now push that branch too.

Still I wonder if the reason why I am getting this error actually lies on the FE:

image

I believe so. I tried in the playground the API with the `userRewardsIncremented` subscription implemented and it worked quite well.

Thank you! 🚀

---

For contributor:

- [x] Added changes entries. Run `yarn changelog` for a guided process.
- [x] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

File: src/components/common/TmBalance.vue [hunk 1]
```diff
@@ -147,6 +146,22 @@ export default {
           // query if successful or not as even an unsuccessful tx costs fees
           refetchNetworkOnly(this.$apollo.queries.overview)
         }
+      },
+      userRewardsIncremented: {
```
Comment (faboweb) [inline src/components/common/TmBalance.vue]: isn't this happening every block if the user has delegations?

Comment (Bitcoinera) [reply src/components/common/TmBalance.vue]: not every block I think, but nearly yes

**Comment by Bitcoinera** [comment]
I am only missing the tests now, like I always do! :sweat_smile:

**Comment by Bitcoinera** [comment]
Can somebody help me out testing the subscriptions? I just noticed they haven't been tested yet throughout the code.

image
:rofl: :rofl: :rofl:

**Comment by faboweb** [comment]
> Can somebody help me out testing the subscriptions? I just noticed they haven't been tested yet throughout the code.

You can just call the functions directly:
`wrapper.vm.$apollo.$subscription.blockAdded.update(result)`

**Comment by Bitcoinera** [comment]
> Why so complicated? There is a TableValidator already does fetch the rewards. Use the blocksubscription there once instead of on every LiValidator?

Fixed! All updating logic now in TableValidators.

**Pull request merged**